[Concept Entry] AI  Diffusion Models #2109 (#2534)
Codecademy/docs: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md

---
Title: 'Diffusion Models'
Description:'Diffusion Models are generative models, which means they are used to generate data similar to what they were trained on. The models work by destroying training data through the addition of Gaussian noise, and then learning to recover that data.' 
Subjects:
  - 'Machine Learning'
  - 'Data Science'
Tags:
  - 'AI'
  - 'Image Generation Model' 
  - 'Denoising'
  - 'Prompts'
  - 'Prompt Engineering'
  - 'Text-To-Image'
  - 'Deep Learning'
CatalogContent:
  - 'machine-learning'
  - 'paths/data-science'
---

**AI Diffusion Models** Are generative models, which means they are used to generate data similar to what they were trained on. The models work by destroying training data through the addition of Gaussian noise, and then learning to recover that data.

> ### About Diffusion Models

Reviewer: ```suggestion
### About Diffusion Models

```
You still need a blank line after this header.

Author: done

> - A Diffusion model is a deep learning generative model that works by adding noise to the available training data known as the forward diffusion process, it then reverses the process to recover the data. This is known as denoising or the reverse diffusion process.
> 
> - The three fundamental mathematical frameworks that support the science behind diffusion models, work on the same principles of adding noise and removing it to generate new samples. 
> 
> - **DDPMs** : Denoising Diffusion Probabilistic Models (generative models mainly used to remove noise from visual or audio data).
> 
> - **SGMs** : Noise-Conditioned Score-Based Generative Models (can generate new samples from a given distribution. They work by learning an estimation score function that can estimate the log density of the target distribution). 
> *In english this means buiding of an estimate based on observed data of which we would like to sample*.
> 
> - **SDEs** : Stochastic Differential Equations (describe changes in random processes concerning time. Widely used in physics and financial markets involving random factors that significantly impact market outcomes).

Reviewer: ```suggestion
A Diffusion model is a deep learning generative model that works by adding noise to the available training data known as the forward diffusion process, it then reverses the process to recover the data. This is known as denoising or the reverse diffusion process.

The three fundamental mathematical frameworks that support the science behind diffusion models, work on the same principles of adding noise and removing it to generate new samples. 

- **DDPMs** : Denoising Diffusion Probabilistic Models (generative models mainly used to remove noise from visual or audio data).
- **SGMs** : Noise-Conditioned Score-Based Generative Models (can generate new samples from a given distribution. They work by learning an estimation score function that can estimate the log density of the target distribution). In english this means buiding of an estimate based on observed data of which we would like to sample.
- **SDEs** : Stochastic Differential Equations (describe changes in random processes concerning time. Widely used in physics and financial markets involving random factors that significantly impact market outcomes).

```
Please follow all the edits in the suggestion. (You can just click on the commit suggestion button.)

Author: done

> ## Image Generation

Reviewer: ```suggestion

```
You still need to get rid of this header.

Author: I had 2 headers with the same name I deleted the other one, there was no edit for this one. Still delete it?

Reviewer: Yes.

> ![Alt text](../../../../../../media/ai-diffusion-model-example.png)

Reviewer: From the instructions on file hosting in the style guide:

> Images should be linked to using the following url: `https://raw.githubusercontent.com/Codecademy/docs/main/media/` followed by the image name.

You also need to put something descriptive in the alt text.

Author: done

Author: I don't see a /main folder to upload too, or maybe im just going about it wrong with the linking

Reviewer: @DevGat718 you upload to the /media folder. That URL above is the URL the image will have after it is merged into the main Docs repo. You can't have a relative link because the URL of the entry when it's on the Doc's website is going to be different than the source on GitHub, but the image is still going to live on the GitHub repo.

Author: copy I put it in media & put in that link lmk if its good.

---
Title: 'Diffusion Models'
Description:'Diffusion Models are generative models, which means they are used to generate data similar to what they were trained on. The models work by destroying training data through the addition of Gaussian noise, and then learning to recover that data.' 
Subjects:
  - 'Machine Learning'
  - 'Data Science'

>   - 'Large Language Model'

Reviewer: ```suggestion
```
Large language model is not one of the subjects in the [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) file.

Author: done

Tags:
  - 'AI'
  - 'Image Generation Model' 
  - 'Denoising'
  - 'Prompts'
  - 'Prompt Engineering'
  - 'Text-To-Image'
  - 'Deep Learning'
CatalogContent:
  - 'machine-learning'
  - 'paths/data-science'
---

**AI Diffusion Models** Are generative models, which means they are used to generate data similar to what they were trained on. The models work by destroying training data through the addition of Gaussian noise, and then learning to recover that data.

> ## More About Diffusion Models

Reviewer: ```suggestion
## About Diffusion Models

```
Blank line after header

Author: done

Author: forgot to add it, can I add it in? @SSwiniarski

> - A Diffusion model is a deep learning generative model that works by adding noise to the available training data known as the forward diffusion process, it then reverses the process to recover the data. This is known as denoising or the reverse diffusion process.
> 
> **3 Diffusion Model Categories**
> - The three fundamental mathematical frameworks that support the science behind diffusion models, work on the same principles of adding noise and removing it to generate new samples. 
> 1. **DDPMs**
> -  Denoising Diffusion Probabilistic Models (generative models mainly used to remove noise from visual or audio data).
> 
> 2. **SGMs**
> - Noise-Conditioned Score-Based Generative Models (can generate new samples from a given distribution. They work by learning an estimation score function that can estimate the log density of the target distribution). 
> *In english this means buiding of an estimate based on observed data of which we would like to sample*.
> 
> 3. **SDEs**
> - Stochastic Differential Equations (describe changes in random processes concerning time. Widely used in physics and financial markets involving random factors that significantly impact market outcomes).

Reviewer: ```suggestion
A Diffusion model is a deep learning generative model that works by adding noise to the available training data known as the forward diffusion process, it then reverses the process to recover the data. This is known as denoising or the reverse diffusion process.

The three fundamental mathematical frameworks that support the science behind diffusion models, work on the same principles of adding noise and removing it to generate new samples.

- **DDPMs** : Denoising Diffusion Probabilistic Models are generative models mainly used to remove noise from visual or audio data.
- **SGMs** : Noise-Conditioned Score-Based Generative Models can generate new samples from a given distribution. They work by learning an estimation score function that can estimate the log density of the target distribution. In English this means building of an estimate based on observed data of which we would like to sample.
- **SDEs** : Stochastic Differential Equations describe changes in random processes concerning time. Widely used in physics and financial markets involving random factors that significantly impact market outcomes.

```
Reformatted this as a single bulleted list.

Author: done

> ### Image Example

Reviewer: ```suggestion
## Image Generation

```

Author: done

> ![Alt text](<ai diffusion model example.png>)

Reviewer: Review the instructions on adding an image in the style guide [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#file-hosting).

Author: copy

Author: done

Author: this was the change

> ## Image Generation

Reviewer: ```suggestion
```
Remove this header.

Author: done

Author: this was the deletion from rd 1
